### PR TITLE
Dev Container and Codespace - fix docker in docker

### PR DESCRIPTION
**Pull request by ilaner**

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -16,12 +16,12 @@ RUN apt-get update && apt-get install dos2unix git python2 curl -y \
     && cd /features \
     # locking to the latest master commit in this repo (https://github.com/devcontainers/features.git) to prevent breaking changes
     # We should update this commit hash from time to time to
```
Comment (dorschw) [inline .devcontainer/Dockerfile:18]:
```suggestion
    # We should update this commit hash from time to time to
```
missing end of sentence 😅

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -16,12 +16,12 @@ RUN apt-get update && apt-get install dos2unix git python2 curl -y \
...
-    && git checkout 96bff0097028001e6e4126c5528d37cb8c13e785
+    && git checkout fc62e9abf47c5ea52e02de997c91c5d52a5edc3a
 
 # This is a workaround for VSCode devcontainer features in self signed certificate
 RUN UID="1000" GID="1000" bash /features/src/common-utils/install.sh
 RUN VERSION="os-provided" bash /features/src/git/install.sh
-RUN VERSION="latest" bash /features/src/docker-in-docker/install.sh
+RUN VERSION="lts" MOBYBUILDXVERSION="0.12.0" bash /features/src/docker-in-docker/install.sh
```
Comment (dorschw) [inline .devcontainer/Dockerfile:24]: why lock here?

Comment (ilaner) [reply .devcontainer/Dockerfile:24]: this is what was broken, from the devcontainer feature docs:

            "description": "Install a specific version of moby-buildx when using Moby. (2024-02-09: Microsoft's Package Manifest has mismatching filesize and SHA for 0.12.1; default is last known good version)"

I'm adding the comment

Comment (dorschw) [reply .devcontainer/Dockerfile:24]: ok please add it

**Review by dorschw — APPROVED** [review]

**Pull request merged**